contracts/bonding/BondingManager.sol

```diff
@@ -548,6 +560,40 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         emit Bond(_to, currentDelegate, _owner, _amount, del.bondedAmount);
     }
 
+    /**
+     * @notice Checkpoints a delegator state after changes, to be used for historical voting power calculations in
+     * on-chain governor logic.
+     */
+    function checkpointBondingState(
+        address _owner,
+        Delegator storage _delegator,
+        Transcoder storage _transcoder
+    ) internal {
+        // start round refers to the round where the checkpointed stake will be active. The actual `startRound` value
+        // in the delegators doesn't get updated on bond or claim earnings though, so we use currentRound() + 1
+        // which is the only guaranteed round where the currently stored stake will be active.
+        uint256 startRound = roundsManager().currentRound() + 1;
+        bondingCheckpoints().checkpointBondingState(
+            _owner,
+            startRound,
+            _delegator.bondedAmount,
+            _delegator.delegateAddress,
+            _delegator.delegatedAmount,
+            _delegator.lastClaimRound,
+            _transcoder.lastRewardRound
+        );
+    }
+
+    /**
+     * @notice Checkpoints the bonding state for a given account.
+     * @dev This is to allow checkpointing an account that has an inconsistent checkpoint with its current state.
+     * Implemented as a deploy utility to checkpoint the existing state when deploying the BondingCheckpoints contract.
+     * @param _account The account to initialize the bonding checkpoint for
+     */
+    function checkpointBondingState(address _account) external {
```

Comment:
Doesn't this allow anyone to trigger checkpointing?

Reply:
Yeah it does, and that's the idea! The checkpoint process is safe, it will only store the current state of the account in the checkpoints contract (now `BondingVotes`). It will be used during state initialization once we deploy this for the first time, and can be used in the future to correct any potential discrepancies that might show up in the checkpointed state compared to the `BondingManager` state. By having this callable for anyone, we also allow people to make sure other addresses have a consistent checkpoint (or even initialize them on the first round).